homeassistant/components/nina/binary_sensor.py

```diff
@@ -69,7 +75,11 @@ def __init__(
 
         self._attr_name = f"Warning: {region_name} {slot_id}"
         self._attr_unique_id = f"{region}-{slot_id}"
-        self._attr_device_class = BinarySensorDeviceClass.SAFETY
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, config_entry.entry_id)},
+            name=f"{config_entry.title}",
```

Comment:
If a device has no name set, it will use the name of the config entry by default. (This will only work if the device is a primary device, thus has a Manufacturer or something else (there's a list of criteria in the device registry somewhere))

Reply:
Thanks for that hint, didn't know that. Remove the `name` attribute, still works and it gets the right name.